clerk/keeper_test.go

```diff
@@ -0,0 +1,124 @@
+package clerk_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/maticnetwork/heimdall/app"
+	"github.com/maticnetwork/heimdall/clerk"
+	"github.com/maticnetwork/heimdall/clerk/types"
+	hmTypes "github.com/maticnetwork/heimdall/types"
+)
+
+//
+// Test suite
+//
+
+// KeeperTestSuite integrate test suite context object
+type KeeperTestSuite struct {
+	suite.Suite
+
+	app *app.HeimdallApp
+	ctx sdk.Context
+}
+
+func (suite *KeeperTestSuite) SetupTest() {
+	suite.app, suite.ctx = createTestApp(false)
+}
+
+func TestKeeperTestSuite(t *testing.T) {
+	suite.Run(t, new(KeeperTestSuite))
+}
+
+//
+// Tests
+//
+
+func (suite *KeeperTestSuite) TestHasGetSetEventRecord() {
```

Comment:
should these be broken into separate tests?

Reply:
i think we should continue with this
getting an event record will involve positive and negative test case - finding the record and not finding
since we have already set once, we can use that to test other cases